Classes/ViewHelpers/Show/ValueViewHelper.php

```diff
@@ -36,12 +36,15 @@ public function render() {
 		$fieldName = $this->templateVariableContainer->get('fieldName');
 
 		if ($value instanceof ObjectStorage) {
-			$object = $value->current();
-
 			// special case for file reference which is ok to be hardcoded
-			if ($object instanceof FileReference) {
-				$value = $object->getOriginalResource()->getName();
+			$objectsArray = $value->toArray();
```

Comment:
We don't have to call $value->toArray() here as the ObjectStorage is iterable.

```
foreach($value as $object) 
```

Can you test so?